src/main/java/spring/intro/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,50 @@
+package spring.intro.dao.impl;
+
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import spring.intro.dao.UserDao;
+import spring.intro.model.User;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
@ Autowired is not necessary above constructor. So either put it on each constructor(with injection) or remove it everywhere